src/components/stack/index.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+const Stack = ({
+	additionalClassNames,
+	alignment,
+	children,
+	direction,
+	divider,
+	flexType,
+	justification,
+	gap,
+	wrap,
+}) => {
+	const defaultAndAdditionalClassnames = `c-stack${additionalClassNames ? ` ${additionalClassNames}` : ''}`;
+	return (
+		<div
+			className={defaultAndAdditionalClassnames}
+			data-style-direction={direction}
+			data-style-justification={justification}
+			data-style-alignment={alignment}
+			data-style-flextype={flexType}
+			data-style-wrap={wrap}
+			style={{ '--c-stack-gap': gap }}
+		>
+			{ Array.isArray(children)
+				? children.map((child) => (
+					<Fragment key={child.key}>
+						{ child }
+						{divider && <hr aria-hidden="true" />}
+					</Fragment>
+				)) : (
+					<>
+						{children}
+						{divider && <hr aria-hidden="true" />}
+					</>
+				)}
+		</div>
+	);
+};
+
+Stack.defaultProps = {
+	additionalClassNames: '',
+	alignment: 'unset',
+	direction: 'vertical',
+	divider: false,
+	flexType: 'flex',
+	justification: 'start',
+	wrap: 'nowrap',
+};
+
+Stack.propTypes = {
+	/** Class name(s) that get appended to default class name of the component */
+	additionalClassNames: PropTypes.string,
+	/** The alignment of the elements within the component.
+	 Relates to CSS Flex's `align-items`
+	 Note: When using the divider in a horizontal configuration, alignment should be set to 'unset' */
+	alignment: PropTypes.oneOf(['unset', 'start', 'center', 'end']),
+	/** Elements that will be displayed within the component. It is recommended that
+	 	each child element have a unique React key attribute */
+	children: PropTypes.any.isRequired,
+	/** The flow of the elements within the component */
+	direction: PropTypes.oneOf(['vertical', 'horizontal']),
+	/** Display a divider between elements within the component.
+	    Note: adding a divider acts as an additional element in the stack
+	 	and effectively doubles the gap applied. Dividers are hidden from screen readers.
+	 	See alignment property when deciding to use the divider. */
+	divider: PropTypes.bool,
+	/** The flex container type */
+	flexType: PropTypes.oneOf(['flex', 'inline-flex']),
```

Comment:
This one feels like it's exposing too much detail about the implementation?  Perhaps a property that defines it as an `inline` (boolean=false) element instead?  Then you can set it as a `inline-flex` vs a default `block` flex.

```
const Stack = ({ inline = false }) => (<div data-style-flextype={inline? 'inline-flex' : 'flex'}>);
```

Reply:
I have been going back and forth with that in my head and I think I want to expose it...perhaps even more (I even think this component should be called Flex).  I want the user to be able to pick it fast and relate it to Flex as much as possible.  I don't see want abstraction gives here for something so closely related to Flex itself....

Reply:
I disagree with calling this component Flex - the responsibilities of a stack component is much more controlled than a Flex component, that's not to say we wouldn't have the need for a Flex component at some point. Trying to make this component do everything flex can is wrong, this components responsibility is for stacking children horizontal or vertically with the ability to add a divider.

Reply:
Updated property from flexType to inline as a bool/